lib/userFactory.js

```diff
@@ -65,6 +65,38 @@ class UserFactory extends BaseFactory {
     }
 
     /**
+     * Get a user model by name, id, or token
+     * @method get
+     * @param   {Mixed}     config
+     * @param   {String}    [config.token]      Token of the user to look up
+     * @return  {Promise}
+     */
+    get(config) {
+        if (!config || !config.token) {
+            return super.get(config);
+        }
+
+        // Lazy load factory dependency to prevent circular dependency issues
+        // https://nodejs.org/api/modules.html#modules_cycles
+        /* eslint-disable global-require */
+        const TokenFactory = require('./tokenFactory');
+        /* eslint-enable global-require */
+        const tokenFactory = TokenFactory.getInstance();
+
+        return tokenFactory.get({ value: config.token })
```

Comment:
All of the hashing is handled by tokens, so I made the factory able to accept `value` which it will then turn into a `hash` param before actually doing the search.

https://github.com/screwdriver-cd/models/blob/user-tokens/lib/tokenFactory.js#L63